core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1966,6 +1967,20 @@ public static TimeZone timeZone(DataContext root) {
     return (TimeZone) DataContext.Variable.TIME_ZONE.get(root);
   }
 
+  /** SQL {@code USER} function. */
+  @NonDeterministic
```

Comment:
why is this non-deterministic?

Reply:
Thanks. Yes this should be deterministic. It was done to be consistent with other functions in the class